testing_utils_test.go

```diff
@@ -142,6 +145,24 @@ func newFakeMarathonEndpoint(t *testing.T, config *Config) *endpoint {
 	}
 }
 
+// basicAuthMiddleware handles basic auth
+func basicAuthMiddleware(username, password string, next http.HandlerFunc) func(http.ResponseWriter, *http.Request) {
+	unauthorized := "{\"message\": \"invalid username or password\"}"
+
+	return func(w http.ResponseWriter, r *http.Request) {
+		u, p, found := r.BasicAuth()
+		if !found && (username != "" || password != "") {
+			http.Error(w, unauthorized, 401)
+			return
+		}
+		if username != u || password != p {
+			http.Error(w, unauthorized, 401)
```

Comment:
FWIW, we would also run into this case if the username and password passed to `basicAuthMiddleware` were both empty but at least one of the credentials parameters contained in the request was not. As described in a comment above, that's actually not how Marathon behaves: It skips authentication if either username and password is empty.

Short-circuiting in the case were either is empty would solve this.